api/src/main/java/org/eclipse/microprofile/config/ConfigProvider.java

```diff
@@ -73,6 +75,16 @@ private ConfigProvider() {
     }
 
     /**
+     * Sets the default config source comparator to define the ordering of auto-discovered
+     * config sources.
+     * @param comparator the new comparator to be used, not null.
+     * @return the previous intance, never null.
+     */
+    public static Comparator<ConfigSource> setDefaultComparator(Comparator<ConfigSource> comparator){
```

Comment:
The comparator can easily sort config sources, if it can identify them, e.g. by class name (all are known in advance).
What else can a comparator rely on in ordering, if it doesn't know anything about the available config sources? 

Some ideas are explained [here](https://github.com/eclipse/microprofile-config/issues/65#issuecomment-282554858) but none is explained in this PR. We need to specify the behavior if the default comparator, if no custom comparator is defined. Otherwise a portable application would be forced to define a comparator.

Apart from checking for an ordinal defined within the config source values or an optional method, we can also define ordering groups, which could cover most use cases (like "Application", "Container", "Environment", "Remote", ...), to make it easier to infer assumptions about config sources.